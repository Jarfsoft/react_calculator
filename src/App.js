function App() {
  return (
    <div>
      React_APP
    </div>
  );
}

export default App;
