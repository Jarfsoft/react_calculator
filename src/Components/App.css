.app {
  display: flex;
  flex-direction: column;
  width: 700px;
  margin-left: auto;
  margin-right: auto;
}
