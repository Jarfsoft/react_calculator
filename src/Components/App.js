import React from 'react';

import Display from './Display';
import ButtonPanel from './ButtonPanel';

function App() {
  return (
    <>
      <Display />
      <ButtonPanel />
    </>
  );
}

export default App;
