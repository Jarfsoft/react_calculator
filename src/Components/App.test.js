import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('Calculator snapshot', () => {
  it('Calculator section is rendered', () => {
    const calculator = renderer
      .create(<App />)
      .toJSON();
    expect(calculator).toMatchSnapshot();
  });
});

describe('Calculation results', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('test adding operator', async () => {
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('60');
    expect(display).toBeInTheDocument();
  });

  it('test substracting operator', async () => {
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('-'));
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('-10');
    expect(display).toBeInTheDocument();
  });

  it('test dividing operator', async () => {
    userEvent.click(screen.getByText('5'));
    userEvent.click(screen.getByText('รท'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('2.5');
    expect(display).toBeInTheDocument();
  });

  it('test multiplying operator', async () => {
    userEvent.click(screen.getByText('7'));
    userEvent.click(screen.getByText('X'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('0'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('140');
    expect(display).toBeInTheDocument();
  });

  it('change number to negative', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('+/-'));
    const display = await screen.findByText('-6');
    expect(display).toBeInTheDocument();
  });

  it('change number to positive', async () => {
    userEvent.click(screen.getByText('-'));
    userEvent.click(screen.getByText('4'));
    userEvent.click(screen.getByText('+/-'));
    const display = await screen.findByText('4');
    expect(display).toBeInTheDocument();
  });

  it('turns number to porcentage', async () => {
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('%'));
    const display = await screen.findByText('0.03');
    expect(display).toBeInTheDocument();
  });

  it('wipe the screen', async () => {
    userEvent.click(screen.getByText('3'));
    userEvent.click(screen.getByText('AC'));
    expect(screen.queryByText('50')).toBeNull();
  });
});
