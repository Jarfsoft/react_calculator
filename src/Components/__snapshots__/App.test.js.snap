// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot Calculator section is rendered 1`] = `
<div
  className="app"
>
  <div
    className="display"
  >
    0
  </div>
  <div
    className="buttonPanel"
  >
    <div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="btn btn3"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="btn btn3"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        X
      </button>
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="btn btn3"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="btn btn3"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="btn"
        onClick={[Function]}
        style={
          Object {
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
